src/formatting/lists.rs

```diff
@@ -734,17 +739,31 @@ pub(crate) fn extract_post_comment(
     // Cleanup post-comment: strip separators and whitespace.
     let post_snippet = post_snippet[..comment_end].trim();
     let post_snippet_trimmed = if post_snippet.starts_with(|c| c == ',' || c == ':') {
-        post_snippet[1..].trim_matches(white_space)
+        post_snippet[1..].trim_matches(white_space).to_string()
     } else if let Some(post_snippet) = post_snippet.strip_prefix(separator) {
-        post_snippet.trim_matches(white_space)
+        post_snippet.trim_matches(white_space).to_string()
     }
     // not comment or over two lines
     else if post_snippet.ends_with(',')
         && (!post_snippet.trim().starts_with("//") || post_snippet.trim().contains('\n'))
     {
-        post_snippet[..(post_snippet.len() - 1)].trim_matches(white_space)
+        post_snippet[..(post_snippet.len() - 1)]
+            .trim_matches(white_space)
+            .to_string()
+    } else if post_snippet.find_uncommented(",").is_some() {
+        let sep_pos = post_snippet.find_uncommented(",").unwrap();
+        [
+            post_snippet[..sep_pos]
+                .trim_matches(white_space)
+                .trim_matches('\n'),
+            "\n",
+            post_snippet[sep_pos + separator.len()..]
+                .trim_matches(white_space)
+                .trim_matches('\n'),
+        ]
+        .concat()
     } else {
-        post_snippet
+        post_snippet.to_string()
```

Comment:
I'd want to avoid always allocating in each branch as it shouldn't be necessary to do so. I know the ownership rules make getting a slice out of the block where there's a `,` nested between comments but there's a few ways to work around this.

One would be to have an unused local mutable variable defined outside the big control flow expression (with an initial String::new assignment to avoid allocating unless and until needed) and then in the nested branch on line 753 that unused local can be assigned to the temp concat string, and the conditional block could then return a slice.

e.g. up above on line ~739 adding a new local (but with a better name :smile:) 
```rust
    // Cleanup post-comment: strip separators and whitespace.
    let post_snippet = post_snippet[..comment_end].trim();
    let mut _nested_comment_case_foo_bar_local = String::new();
    let post_snippet_trimmed = if post_snippet.starts_with(|c| c == ',' || c == ':') {
```

and then use that `_nested_comment_case_foo_bar_local` in this branch:

```rust
        _nested_comment_case = [
            post_snippet[..sep_pos]
                .trim_matches(white_space)
                .trim_matches('\n'),
            "\n",
            post_snippet[sep_pos + separator.len()..]
                .trim_matches(white_space)
                .trim_matches('\n'),
        ]
        .concat();

        &_nested_comment_case_foo_bar_local
```

Perhaps another approach could be to have the conditional return a tuple that would contain the array of slices and a boolean indicator as to whether the entries need to be concatenated

Reply:
Wow! I was unaware of this tactical approach to solving ownership problems. Thanks for sharing it.  :heart: